app/models/draft_content_item.rb

```diff
@@ -35,6 +35,29 @@ class DraftContentItem < ActiveRecord::Base
   }
   validates_with RoutesAndRedirectsValidator
 
+  validates :description, well_formed_content_types: { must_include: "text/html" }
+  validates :details, well_formed_content_types: { must_include: "text/html" }
+
+  # Postgres's JSON columns have problems storing literal strings, so these
+  # getter/setter overrides wrap and unwrap those strings in hashes.
+  def description=(value)
+    if value.is_a?(String)
+      super(string: value)
+    else
+      super
+    end
+  end
+
+  def description
+    value = super
+
+    if value.is_a?(Hash) && value.key?(:string)
+      value.fetch(:string)
+    else
+      value
+    end
+  end
```

Comment:
I might be inclined to remove the type check:

```ruby
def description=(value)
  super(string: value.to_s)
end

def description
  super.fetch(:string)
end
```

You lose some correctness for cases where the description is being set as a properly constructed hash already, or it was somehow stored as an incorrect hash.  But, the api of the object is that the description attribute is a string and the content item will store that appropriately - if you give it anything else it'll get massaged into a shape we expect (e.g. what AR would do with any other field that you give it the wrong type of object - it'll do a simple coercion and serialize away).